cypress/e2e/ConnectWallet.cy.ts

```diff
@@ -5,57 +5,83 @@ describe('Connect Wallet', () => {
 	});
 
 	afterEach(() => {
-		// reset connection for following test
+		// disconnect wallet for following test
 		cy.disconnectMetamaskWalletFromDapp().then((disconnected) => {
 			expect(disconnected).to.be.true;
 		});
 	});
 
 	after(() => {
-		cy.resetMetamaskAccount();
+		// reset account after test suite
+		cy.resetMetamaskAccount().then((reset) => {
+			expect(reset).to.be.true;
+		});
 	});
 
-	it('successfully opens the connect wallet modal', () => {
+	// fire click event on connect button
+	const clickConnectBtn = () =>
 		cy.get('.authentication__connect-wrapper').click();
-		cy.get('div').should('have.class', 'wallet-select__header');
-		cy.contains('connect to a wallet', { matchCase: false }).should(
-			'be.visible',
-		);
-		cy.get('.dialog__content')
-			.should('contain', 'Metamask')
-			.and('contain', 'Wallet Connect')
-			.and('contain', 'Coinbase Wallet')
-			.and('contain', 'Fortmatic')
-			.and('contain', 'Portis');
+
+	it('successfully opens the connect wallet modal', () => {
+		clickConnectBtn();
+
+		// assert find connect wallet modal
+		cy.findByText('Connect To A Wallet').should('be.visible');
+
+		// assert connect wallet provider options
+		cy.findByText('Metamask').should('be.visible');
+		cy.findByText('Wallet Connect').should('be.visible');
+		cy.findByText('Coinbase Wallet').should('be.visible');
+		cy.findByText('Fortmatic').should('be.visible');
+		cy.findByText('Portis').should('be.visible');
 	});
 
 	it('displays connecting indicator when list item is clicked', () => {
-		cy.get('.authentication__connect-wrapper').click();
+		clickConnectBtn();
+
+		// assert and click metamask wallet provider option
 		cy.findByText('Metamask').trigger('mouseover').click();
+
 		cy.wait(2000);
+
 		cy.get('.dialog__content').contains('connecting...', {
 			matchCase: false,
 		});
+
```

Comment:
Is there an equivalent of `waitUntil` here? Just so it doesn't have to wait `2000ms`. On quick scan, it looks like `.should` retries for up to `4s`, so that could be a good option here!

Reply:
Yeah `wait` isn't required here if `should` is used for chaining. Thanks for spotting that 👍